bibliosearch_backend/api/views.py

```diff
@@ -1,3 +1,101 @@
+import json
 from django.shortcuts import render
 
+#import auth
+from django.contrib.auth import authenticate
+
+from django.contrib.auth import logout as auth_logout
+
+from django.contrib.auth import login as auth_login
+
+from django.http import JsonResponse
+
+#import messages
+from django.contrib import messages
+
+#import redirect
+from django.shortcuts import redirect
+
+#import User
+from django.contrib.auth.models import User
+
+#import BiblioSearchUser
+from api.models import BiblioSearchUser
+
+from django.views.decorators.csrf import csrf_exempt
+
+
+#import decorators
+from django.contrib.auth.decorators import login_required  
+
+
 # Create your views here.
+
+# create login view
+@csrf_exempt
+def login(request):
+    if request.method == 'POST':
+        data = json.loads(request.body)
+        username = data['username']
+        password = data['password']
+        user = authenticate(request, username=username, password=password)
```

Comment:
instead of if check you can use 


```python
from django.views.decorators.http import require_http_methods

@require_http_methods(["POST"])
def login(request):
    ...
```

this will only allow POST requests to this view, and will return proper errors to the client.